abstract/encoding.go

```diff
@@ -292,8 +292,10 @@ func (en *encoder) value(obj interface{}, depth int) error {
 		}
 
 	case reflect.Int:
-		i := int64(v.Int())
-		return binary.Write(en.w, binary.BigEndian, i)
+		t := reflect.TypeOf(int32(0))
```

Comment:
On second thought, I think panic is indeed what we want or encoding errors like this where the code calling the encoder incorrectly formed the object to be encoded, so never mind that part of my comment above.  (The need for the err return is mainly to pass back errors from Write() calls cleanly, which can happen due to external conditions, e.g., disk full or network connection dropped, and not just due to internal bugs.  So there's still a legit, and different, need, for that error return path.)

But the considerations about unnecessarily using expensive reflection remain. :)